core/src/main/java/com/yahoo/imapnio/async/request/AbstractFetchCommand.java

```diff
@@ -54,66 +96,144 @@
+    }
+
+    /**
+     * Initializes a {@link AbstractFetchCommand} with the flag indicate whether prepending UID, the message numbers string, the data items,
+     * changed since the given modification sequence, and the flag to check whether uid fetch with vanished option.
+     *
+     * @param isUid whether prepending UID
+     * @param msgNumbers the message numbers string
+     * @param items the data items
+     * @param changedSince changed since the given modification sequence
+     * @param isVanished whether uid fetch with vanished option
+     */
+    protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNumbers, @Nonnull final String items,
+                                   @Nullable final Long changedSince, final boolean isVanished) {
         this.isUid = isUid;
         this.msgNumbers = msgNumbers;
         this.dataItems = items;
         this.macro = null;
+        this.changedSince = changedSince;
+        this.isVanished = isVanished;
     }
 
     /**
-     * Initializes a {@link FetchCommand} with the {@link MessageNumberSet} array.
+     * Initializes a {@link AbstractFetchCommand} with the flag indicate whether prepending UID, the message numbers string, the macro,
+     * changed since the given modification sequence, and the flag to check whether uid fetch with vanished option.
      *
      * @param isUid whether prepending UID
      * @param msgNumbers the message numbers string
      * @param macro the macro
+     * @param changedSince changed since the given modification sequence
+     * @param isVanished whether uid fetch with vanished option
      */
-    protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNumbers, @Nonnull final FetchMacro macro) {
+    protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNumbers, @Nonnull final FetchMacro macro,
+                                   @Nullable final Long changedSince, final boolean isVanished) {
         this.isUid = isUid;
         this.msgNumbers = msgNumbers;
         this.macro = macro;
         this.dataItems = null;
+        this.changedSince = changedSince;
+        this.isVanished = isVanished;
     }
 
     @Override
     public void cleanup() {
         this.msgNumbers = null;
         this.dataItems = null;
         this.macro = null;
+        this.changedSince = null;
     }
 
     @Override
     public ByteBuf getCommandLineBytes() {
-        final ByteBuf sb = Unpooled.buffer();
+        // Ex:UID FETCH 300:500 (FLAGS) (CHANGEDSINCE 1234 VANISHED)
+        final int dataItemsSize = dataItems == null ? macro.name().length() : dataItems.length();
+        final String changedSinceStr = changedSince == null ? "" : changedSince.toString();
+        final int vanishedSize = isVanished ? SP_VANISHED.length() : 0;
+        final int len = ImapClientConstants.PAD_LEN  + changedSinceStr.length() + dataItemsSize + msgNumbers.length() + vanishedSize;
```

Comment:
Vanished is only used for UID fetch. Even though abstract constructor has changed since parameter nullable for clean clean implementation of constructors overloading, I enforce UID fetch constructor to require changed since to be non null when having is vanished parameter.